### ISSUE TITLE
Need to have optimizer function to train with gradients

### ISSUE THREAD

**Issue by verystrongjoe** [keras-team/keras#10033]
Hi, all!

I am building a3c model based on [this paper](https://arxiv.org/abs/1602.01783), named "asynchronous methods for deep reinforcement learning".  I uploaded my source code on [here](https://github.com/verystrongjoe/a3c-sketch).

I will give you a brief situation of mine. I already built a global and a local agent where those agents can run in other servers communicating web socket. Because the environment that I am running is too heavy to load it in each thread.  Anyway, I get gradients of local agent which will send it to its global agent and then global agent will gather all gradients and accumulate them. Then I have a problem. I can't how to train with this to global network.

In tensorflow, there is apply_gradients() method in optimizer like [here](https://stackoverflow.com/questions/42156957/how-to-update-model-parameters-with-accumulated-gradients).

Please add this function to optimizer class if it is possible.
